cmd/insights/contributors.go

```diff
@@ -134,11 +136,29 @@ func (cis contributorsInsightsSlice) BuildOutput(format string) (string, error)
 		return utils.OutputJSON(cis)
 	case constants.OutputYAML:
 		return utils.OutputYAML(cis)
+	case constants.OuputCSV:
+		return cis.OutputCSV()
 	default:
 		return "", fmt.Errorf("unknown output format %s", format)
 	}
 }
 
+func (cis contributorsInsightsSlice) OutputCSV() (string, error) {
+	if len(cis) == 0 {
+		return "", fmt.Errorf("repository is either non-existent or has not been indexed yet")
+	}
+	b := new(bytes.Buffer)
+	writer := csv.NewWriter(b)
+	err := writer.WriteAll([][]string{{"Repository URl", "New Contributors", "Recent Contributors", "Alumni Contributors", "Repeat Contributors"}, {
+		cis[0].RepoURL, strconv.Itoa(len(cis[0].New)), strconv.Itoa(len(cis[0].Recent)), strconv.Itoa(len(cis[0].Alumni)), strconv.Itoa(len(cis[0].Repeat)),
```

Comment:
This breaks with multiple `contributorsInsightsSlice` since it only ever takes the first one via index `0`.

Notice that when I run this with multiple repos, I get only one:
```
❯ pizza insights contributors https://github.com/open-sauced/api https://github.com/open-sauced/app -o csv
Repository URl,New Contributors,Recent Contributors,Alumni Contributors,Repeat Contributors
https://github.com/open-sauced/app,21,28,7,7

❯ pizza insights contributors https://github.com/open-sauced/api https://github.com/open-sauced/app -o csv
Repository URl,New Contributors,Recent Contributors,Alumni Contributors,Repeat Contributors
https://github.com/open-sauced/api,3,7,3,4

❯ pizza insights contributors https://github.com/open-sauced/api https://github.com/open-sauced/app -o csv
Repository URl,New Contributors,Recent Contributors,Alumni Contributors,Repeat Contributors
https://github.com/open-sauced/api,3,7,3,4

❯ pizza insights contributors https://github.com/open-sauced/api https://github.com/open-sauced/app -o csv
Repository URl,New Contributors,Recent Contributors,Alumni Contributors,Repeat Contributors
https://github.com/open-sauced/api,3,7,3,4

❯ pizza insights contributors https://github.com/open-sauced/api https://github.com/open-sauced/app -o csv
Repository URl,New Contributors,Recent Contributors,Alumni Contributors,Repeat Contributors
https://github.com/open-sauced/app,21,28,7,7
```

(and since the API doesn't return them ordered, it's never consistent which is the one being printed).

Can we instead first write the header for the csv (the `Repository URL`, `New Contributors`, etc.) and then enter a loop to write each of the `cis` within the slice?

Reply:
Looks great!

```
❯ pizza insights contributors https://github.com/open-sauced/api https://github.com/open-sauced/app -o csv
Repository URL,New Contributors,Recent Contributors,Alumni Contributors,Repeat Contributors
https://github.com/open-sauced/api,3,7,3,4
https://github.com/open-sauced/app,22,29,7,7
```